### DISCUSSION TITLE
v0.5 Env (Candidate)

### DISCUSSION THREAD

**Discussion by matthewmueller** [livebud/bud#21] [Announcements]
## Goal

Bud is simpler to deploy to a PaaS like Heroku and Fly.io because it's a stock Go application that longer depends on Node.js. We're taking inspiration from Rails here which recently switched off Node.js in favor of import maps.

## Deliverable

Demo that rebuilds the minimal HN clone, but without Node and deployed to Fly.io.

## Tasks

```mermaid
graph BT
classDef done fill:green,color:white,stroke:green;
classDef wip fill:yellow,stroke:yellow;
1(Add env generator)
2(Support import maps)
3(Expose DI type mapping)
4(Support multiple environments)
5(Remove Node.js)
6(Rehearse Demo)
7(Release v0.3)
8(Record Demo)
1-->2
1-->3
1-->4
2-->5
3-->6
4-->6
5-->6
6-->7
7-->8
```

**Comment by matthewmueller** [comment]
Added an RFC for [supporting Import Maps](https://github.com/livebud/bud/discussions/74). Feedback welcome!

**Comment by diyism** [comment]
The plenti project has successfully removed nodejs since 2020-08, how soon will the bud v0.3 (without nodejs) be released? I'm very much expecting the bud v0.3.

**Reply by matthewmueller** [reply]
A slight nuance is that Bud doesn't use Node.js today, it uses V8. We do still rely on the svelte and livebud packages to be in `node_modules/`, but Bud doesn't care how you get them there. 

I'm working on improving the viewer with: https://github.com/livebud/bud/pull/401. I'm going to try to make node_modules optional, but I think in practice, most people will want to install dependencies this way.

What do you think @diyism ?

**Reply by diyism** [reply]
I'm confused, are you saying the bud v0.2.8 has removed the Node v14+ requirement? but the project Readme.md still has it:
![Screenshot 2023-04-09 at 13-43-48 livebud_bud The Full-Stack Web Framework for Go](https://user-images.githubusercontent.com/1744399/230756552-0b639473-5523-4cc5-9ddc-aa83642d7ce6.png)

**Comment by diyism** [comment]
I found svelte-bun which realized realtime building of svelte components on  dev environment.
But Bun is not on golang.

ref: https://github.com/xyproto/algernon/issues/125

**Comment by diyism** [comment]
I see you've replaced nodejs with v8go, so the "Node v14+" can be deleted from the Readme.md?
